libraries/node-core-library/src/JsonSchema.ts

```diff
@@ -180,9 +184,9 @@ export class JsonSchema {
 
       buffer += os.EOL + indent + `       ${errorDetail.message}`;
 
-      if (errorDetail.inner) {
-        buffer = JsonSchema._formatErrorDetailsHelper(errorDetail.inner, indent + '  ', buffer);
-      }
+      // if (errorDetail.inner) {
+      //   buffer = JsonSchema._formatErrorDetailsHelper(errorDetail.inner, indent + '  ', buffer);
+      // }
```

Comment:
Not sure where in Ajv `inner` [would be nested, **Ajv `ValidationError` does not have a nested tree**](https://github.com/ajv-validator/ajv/blob/v6.12.3/lib/compile/error_classes.js)

Reply:
Consider a schema that allows possibilities like this:

```
option 1:
- kind=square
- size=small

option 2:
- kind=square
- size=medium

option 3:
- kind=circle
- size=medium

option 4:
- kind=circle
- size=large
```

And then suppose your input is `{ kind=square, size=large }`.  Then the error message has to say something like:

- I tried matching `kind=square`, but in that case the size needs to be `small` or `medium`
- I also tried matching `size=large`, but in that case your kind needs to be `circle`

The reason is that the `any of` and `all of` constructs don't always have a way to tell the engine that `kind` is a special important property and `size` is a secondary detail.  The schema just sees possible patterns, so the error message often won't be understandable without more explanation of how it tried to match it.  And this problem can get nested arbitrarily in complex schemas, so it ends up being a tree.

I know that the newer JSON Schema supports mechanisms like inheritance that make this problem better. And `ajv` itself might have better ways of reporting error messages. If so -- we should try to tailor `JsonSchema` around that rather than forcing it to use z-schema's design. The goal is informative error messages.

Reply:
(If I can find some more time, I'll see if I can dig up an actual test case that better illustrates a case where nesting error messages were helpful.)

Reply:
Yes, at least one schema, and a few fixtures where we know what errors should throw.
Not the shape of the error, known and easy to spot fixture where we could see quickly where.